### PR TITLE
Refs #30842 - Update Stories files for Storybook v6 upgrade

### PR DESCRIPTION
**Pull request by jeremylenz**
This PR contains the changes to Stories files needed after the upgrade to Storybook v6.

* Use new [component names](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#previewprops-renamed): `Preview` -> `Canvas`; `Props` -> `ArgsTable`
* Change the [heirarchy separator](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#removed-hierarchy-separators) from `|` to `/`
* Import Storybook addon packages directly, not from `@theforeman/stories` (reason why and discussion on https://github.com/theforeman/foreman-js/pull/214)
* adjust some story ordering / cleanup

### To test:

* `cd ../foreman` and checkout this PR
* `npm run stories`
* Verify that stories are rendering
* Verify no console errors
* Change a stories file that has
```js
import {action} from '@storybook/addon-actions';
```
to
```js
import { action } from '@theforeman/stories';
```
You should see the helpful deprecation error.

### CODE REVIEW

**Comment by jeremylenz** [comment]
update: I think the reason for the MDX weirdness was that `@mdx-js/loader` was being run twice on the same files, once by the Storybook preset and once by our manual webpack configuration.

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.stories.js
Comment (jeremylenz) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbBar.stories.js:3]: these eslint errors should go away once the foreman-js PR is merged

**Comment by johnpmitsch** [comment]
Looks good and working well! 

Looks like FactChartModal still needs to be updated, I'm seeing a deprecation warning there.

The other components are rendering fine, seeing some warnings with some of them. Not sure if it's pre-existing issues, probably not storybook related but maybe could see if any are easy fixes:

- Charts > ChartBox
```
Warning: Failed prop type: The prop `type` is marked as required in `ChartBox`, but its value is `undefined`.
    in ChartBox (created by storyFn)
    in storyFn
    in ErrorBoundary
```
- Common > Action Buttons
```
Warning: Failed prop type: The prop `id` is required to make `Dropdown` accessible for users of assistive technologies such as screen readers.
```
- Common > Orderable: 
```
Failed prop type: Invalid prop `options` of type `array` supplied to `OrderAppSandbox`, expected `object`.
```
- Bookmarks > BookmarkForm
```
Warning: Failed prop type: The prop `setModalClosed` is marked as required in `BookmarkForm`, but its value is `undefined`.
```

**Comment by jeremylenz** [comment]
@johnpmitsch rebased.  I would like to leave the above updates for the next PR since I don't think they're upgrade-related.

**Comment by jeremylenz** [comment]
JavaScript test failures will be resolved once https://github.com/theforeman/foreman-js/pull/214 is merged (see https://github.com/theforeman/foreman-js/pull/214/commits/885d9bbb7009808a3e51012f6070799936549c02)

**Comment by johnpmitsch** [comment]
@jeremylenz still working well, just one story looks like it's not working, "Use Memory Allocation Input" (at the bottom), seeing `Error: text is no longer exported from @theforeman/stories.  Import it directly from @storybook/addon-knobs.`

**Comment by jeremylenz** [comment]
@johnpmitsch Thanks, updated

File: webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.stories.js
Comment (johnpmitsch) [inline webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.stories.js]:
```suggestion
import { number, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
```
Looks like action needs to come from here! Once I made this change it all worked fine :+1:

**Review by johnpmitsch — CHANGES_REQUESTED** [review]
Just the one import change!

File: webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.stories.js
Comment (jeremylenz) [reply webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.stories.js]: oops!  :yum:

**Comment by jeremylenz** [comment]
@johnpmitsch updated :)

**Review by johnpmitsch — DISMISSED** [review]
APJ

**Comment by MariaAga** [comment]
Thanks Jeremy for the upgrade! 
Is it possible to fix the knobs in this PR? They don't show up for me with or without this PR...
Also missing one title change here: `Components|Form/MemoryAllocationInput` :) 
file location:
`webpack/assets/javascripts/react_app/components/MemoryAllocationInput/MemoryAllocationInput.stories.js`

**Comment by jeremylenz** [comment]
@MariaAga thanks, missed that one!

I updated the stories that use knobs, since I realized you no longer have to specify the `withKnobs` decorator, per [this part of the migration guide](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-addon-presets).  However, for me it doesn't seem to break anything either way.  All the stories I tried that use knobs seem to work fine.  Can you point me to a particular one that's not working?

**Comment by MariaAga** [comment]
I tried with Firefox and it looks like the knobs are showing up... It looks like a weird css issue as the elements are there in chrome just hidden so it's probably an issue with my env. I'm glad to hear that it's working for you :)
Clicking on "Change addons orientation" somehow fixed it :tada: 
Screenshot from 2020-11-22 10-50-48

**Comment by jeremylenz** [comment]
[test foreman]

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
[test katello]

File: webpack/stories/docs/slot-and-fill.stories.mdx
Comment (ekohl) [inline webpack/stories/docs/slot-and-fill.stories.mdx]: Why this change? This makes it look like you always need to use `:plugin` while it's really dependent on what you use.

Comment (jeremylenz) [reply webpack/stories/docs/slot-and-fill.stories.mdx]: @ekohl I think that's left over from when I was troubleshooting the [weirdness](https://github.com/theforeman/foreman/pull/7990#issuecomment-700238606) with mdx-loader trying to parse the same file twice.  I will change it back.

File: webpack/stories/docs/slot-and-fill.stories.mdx
Comment (ekohl) [inline webpack/stories/docs/slot-and-fill.stories.mdx]:
```suggestion
Foreman::Plugin.register :<plugin> do
```

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
[test katello]

**Comment by jeremylenz** [comment]
@ekohl @MariaAga @johnpmitsch all checks passing, let me know if there are any more concerns? :)

**Comment by johnpmitsch** [comment]
@jeremylenz my ack still stands :+1:

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @jeremylenz ! :+1: 

I found two regressions:
1. The copy button for copying snippets doesn't work
Introduction _ Getting Started - Page ⋅ Storybook

2. The `BreadcrumbBar` story shows a blank page
blank-breadcrumbs

**Comment by MariaAga** [comment]
About the breadcrumbs: the PR might need a rebase as I solved an issue in breadcrumbs sometime ago: https://github.com/theforeman/foreman/pull/8146

**Comment by jeremylenz** [comment]
After rebase the breadcrumb bar story shows correctly for me:

breadcrumb_bar_story

**Comment by jeremylenz** [comment]
@amirfefer I think the copy issue is because the new version of `addon-docs` uses the browser `navigator.clipboard` API which [requires https](https://web.dev/async-clipboard/#security-and-permissions), and the local Storybook runs http.  The issue should go away in the deployed storybook because you can use https.  You aren't seeing the issue in the old storybook because `addon-docs` [recently changed](https://github.com/storybookjs/storybook/pull/11251) from the older clipboard API to the new one.

**Comment by jeremylenz** [comment]
@MariaAga @amirfefer Rebased and also fixed one straggling hierarchy separator in the Client Routing story.

**Review by amirfefer — APPROVED** [review]
Thanks @jeremylenz :+1: 
LGTM

**Comment by jeremylenz** [comment]
@ekohl @sharvit any packaging concerns?

**Review by amirfefer — CHANGES_REQUESTED** [review]
@jeremylenz, this [story](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/SkeletonLoader/SkeletonLoader.stories.js#L11) has just been added, could you please update it?

This also needs an ACK from @theforeman/packaging

**Comment by jeremylenz** [comment]
@amirfefer thanks, rebased and updated

**Comment by jeremylenz** [comment]
@amirfefer Rebased again to fix `package.json` merge conflict

**Comment by jeremylenz** [comment]
@theforeman/packaging any concerns here?

**Comment by ezr-ondrej** [comment]
> @theforeman/packaging any concerns here?

I think it's just devDependency, so it should not need to be packaged? :thinking:

**Review by tbrisker — APPROVED** [review]
Ack packaging wise, this module isn't packaged. @amirfefer feel free to merge if all your comments were addressed.

**Comment by evgeni** [comment]
> > @theforeman/packaging any concerns here?
> 
> I think it's just devDependency, so it should not need to be packaged? :thinking: 

This isn't true, some devDeps we need packaged, but this one not.

**Review by evgeni — COMMENTED** [review]
packaging 👻 says yay

**Review by evgeni — APPROVED** [review]

**Comment by jeremylenz** [comment]
[test katello]

**Review by amirfefer — APPROVED** [review]
Thanks @jeremylenz !
Tested and works well 👍 
There's an unrelated regression with `api-middleware-usage.stories.mdx` 
```
WARNING in ./webpack/stories/docs/api-middleware-usage.stories.mdx
Module build failed (from ./node_modules/@theforeman/stories/node_modules/@mdx-js/loader/index.js):
SyntaxError: Unexpected token (7:0)
```
This fails because of the  `useAPI` addition - https://github.com/theforeman/foreman/pull/8153/files#diff-be6f7f254ad11504ed27e7d0d01b6bc5626f9adc0585abc02574c32d2890b2fdR9

I created a fix for it - https://github.com/theforeman/foreman/pull/8223

**Pull request merged**
